pkg/cmd/dockerregistry/dockerregistry.go

```diff
@@ -137,7 +138,15 @@ func Execute(configFile io.Reader) {
 	app.Config.HTTP.Headers.Set("X-Registry-Supports-Signatures", "1")
 
 	app.RegisterHealthChecks()
-	handler := alive("/", app)
+	handler := http.Handler(app)
```

Comment:
Also, we already have a crude pullthrough limiter that limits max simultaneous writes.